runtime/near-vm-logic/src/logic.rs

```diff
@@ -162,6 +163,13 @@ impl<'a> VMLogic<'a> {
         if self.config.regular_op_cost == 0 {
             return Ok(());
         }
+        if !checked_feature!(
+            "protocol_feature_wasm_global_gas_counter",
```

Comment:
Did we decide to do this through a protocol upgrade @matklad ?

Reply:
Yes, discussed with @matklad and @nearmax this is a protocol change. [This comment](https://github.com/near/nearcore/pull/4688#issuecomment-906991528) described why.

For this specific line: in old binary only old protocol version code available, this function does not exist. In new binary, both new and old counter logic exists, which to choose depends on current epoch protocol version passed from caller functions. sync_from/to_wasm_counter is injected to each host function, in the case of old protocol version, they're skipped by this if.